provisiondomainmanager_policy.tf

```diff
@@ -0,0 +1,527 @@
+# ------------------------------------------------------------------------------
+# Create the IAM policies that allow sufficient permissions to provision
+# all AWS resources for Domain Manager in the User Services account.
+# ------------------------------------------------------------------------------
+
+data "aws_iam_policy_document" "provisiondomainmanager_policy_acm_cognito_ec2_doc" {
+  statement {
+    actions = [
+      "acm:DeleteCertificate",
+      "acm:GetCertificate",
+      "acm:ListTagsForCertificate",
+      "acm:RemoveTagsFromCertificate",
+      "acm:UpdateCertificateOptions",
+      "cognito-idp:AdminAddUserToGroup",
+      "cognito-idp:AdminConfirmSignUp",
+      "cognito-idp:AdminCreateUser",
+      "cognito-idp:AdminDeleteUser",
+      "cognito-idp:AdminDisableUser",
+      "cognito-idp:AdminEnableUser",
+      "cognito-idp:AdminGetUser",
+      "cognito-idp:AdminListDevices",
+      "cognito-idp:AdminListGroupsForUser",
+      "cognito-idp:AdminRemoveUserFromGroup",
+      "cognito-idp:AdminResetUserPassword",
+      "cognito-idp:ConfirmSignUp",
+      "cognito-idp:CreateGroup",
+      "cognito-idp:CreateUserPoolClient",
+      "cognito-idp:CreateUserPoolDomain",
+      "cognito-idp:DeleteGroup",
+      "cognito-idp:DeleteUser",
+      "cognito-idp:DeleteUserPool",
+      "cognito-idp:DeleteUserPoolClient",
+      "cognito-idp:DeleteUserPoolDomain",
+      "cognito-idp:GetGroup",
+      "cognito-idp:GetUser",
+      "cognito-idp:GetUserPoolMfaConfig",
+      "cognito-idp:ListGroups",
+      "cognito-idp:ListUserPoolClients",
+      "cognito-idp:ListUsers",
+      "cognito-idp:ListUsersInGroup",
+      "cognito-idp:SetUserPoolMfaConfig",
+      "cognito-idp:SignUp",
+      "cognito-idp:UpdateGroup",
+      "cognito-idp:UpdateUserPool",
+      "cognito-idp:UpdateUserPoolClient",
+      "cognito-idp:UpdateUserPoolDomain",
+      "ec2:AuthorizeSecurityGroupEgress",
+      "ec2:AuthorizeSecurityGroupIngress",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteTags",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RevokeSecurityGroupIngress",
+      "ec2:UpdateSecurityGroupRuleDescriptionsEgress",
+      "ec2:UpdateSecurityGroupRuleDescriptionsIngress"
+    ]
+
+    resources = ["*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:ResourceTag/app"
+      values   = ["domain-manager"]
+    }
```

Comment:
Can we use this sort of foo on any of the other `statement`s?

Reply:
Good question- I need @zenine07 to answer that.

Reply:
It depends on the service. Some services support it, others do not. We have it setup so that the services which we can use ARNs for, are using arns, then those that we cannot use ARNs for, are using tags. I've tried adding those condition based to other statements, but most don't play super nicely with it. The ARNs are a lot easier if you have a good naming convention.

You can also add aws:RequestTag as a condition for when creating resources, but not all resources play nicely with that, as tags are added after creating the resource. Hence some statements allowing create for all resources.

Reply:
Thanks @zenine07 - appreciate your reply.  Please keep an eye out for any other places where we can tag resources to help us enforce least-privilege permissions (across all of our repositories).

Reply:
No problem. If you want to see what resources/services support which condition keys, you can go here https://docs.aws.amazon.com/service-authorization/latest/reference/reference_policies_actions-resources-contextkeys.html